packages/testing-kit/src/wallet/lucid/provisionning.ts

```diff
@@ -0,0 +1,181 @@
+    }
+    const { lucid, wallet } = di;
+    let requestWithWallets: RequestWithWallets = {};
+    await Promise.all(
+        Object.entries(request).map(([participant, {walletSeedPhrase, scheme}]) => {
+        return Lucid
+            .new(lucid.provider,lucid.network)
+            .then((newLucidInstance) => {
+                const wallet = mkLucidWalletTest(newLucidInstance.selectWalletFromSeed(walletSeedPhrase.join(` `)));
+                requestWithWallets[participant] = { wallet, scheme };
+            });
+        })
+    );
+
+    logInfo(
+        `Provisionning Request : ${MarloweJSON.stringify(
+        requestWithWallets,
+        null,
+        4
+        )}`
+    );
+    
+    const mintingDeadline = addDays(Date.now(), 1);
+
+    const [script, policyId] = await mkPolicyWithDeadlineAndOneAuthorizedSigner(
+      di
+    )(mintingDeadline);
+
+    const distributions = await Promise.all(
+      Object.entries(requestWithWallets).map(([participant, x]) =>
+        x.wallet.getChangeAddress().then(
+          (address) =>
+            [
+              participant,
+              x.wallet,
+              RuntimeCore.addressBech32(address),
+              {
+                lovelaces: x.scheme.lovelacesToTransfer,
+                tokens: Object.entries(x.scheme.assetsToMint).map(
+                  ([assetName, quantity]) => ({
+                    quantity,
+                    assetId: { assetName, policyId },
+                  })
+                ),
+              },
+            ] as [
+              string,
+              WalletTestAPI,
+              RuntimeCore.AddressBech32,
+              RuntimeCore.Assets
+            ]
+        )
+      )
+    );
+
+    const assetsToMint = pipe(
+      distributions,
+      A.map((aDistribution) => toAssetsToMint(aDistribution[3])),
+      A.reduce(mergeAssets.empty, mergeAssets.concat)
+    );
+
+    logDebug(`Distribution : ${MarloweJSON.stringify(distributions,null,4)}`);
+    logDebug(`Assets to mint : ${MarloweJSON.stringify(assetsToMint,null,4)}`);
+
+    const mintTx = lucid
+      .newTx()
+      .mintAssets(assetsToMint)
+      .validTo(Date.now() + 100000)
+      .attachMintingPolicy(script);
+
+    const transferTx = await pipe(
+      distributions,
+      A.reduce(
+        lucid.newTx(),
+        (tx, aDistribution) =>
+          tx
+            .payToAddress(
+              aDistribution[2],
+              toAssetsToTransfer(aDistribution[3])
+            )
+            .payToAddress(aDistribution[2], { lovelace: 5_000_000n })
+            .payToAddress(aDistribution[2], { lovelace: 5_000_000n })
+            .payToAddress(aDistribution[2], { lovelace: 5_000_000n }) // add a Collateral
+      )
+    );
+
+    var result: ProvisionResponse = {};
+    distributions.map(([participant, wallet, , assetsProvisionned]) => {
+      result[participant] = {
+        wallet: wallet,
+        assetsProvisionned: assetsProvisionned,
+      };
+    });
```

Comment:
We don't need to mutate at all actually
```suggestion
    const result = Object.fromEntries(
      distributions.map(([participant, wallet, , assetsProvisionned]) =>
        [participant, { wallet, assetsProvisionned }]));
```

Also `assetsProvisionned` should be `assetsProvisioned`.